crates/bevy_ecs/src/component.rs

```diff
@@ -173,13 +173,25 @@ pub struct ComponentDescriptor {
     is_send_and_sync: bool,
     type_id: Option<TypeId>,
     layout: Layout,
-    drop: unsafe fn(*mut u8),
+    drop: for<'a> unsafe fn(OwningPtr<'a>),
+}
+
+impl std::fmt::Debug for ComponentDescriptor {
```

Comment:
```suggestion
// We need to ignore the `drop` field in our `Debug` impl
impl std::fmt::Debug for ComponentDescriptor {
```

Reply:
Why is that necessary? And can we still get the field of we cast to a raw pointer?